components/create_post/create_post.test.jsx

```diff
@@ -295,6 +297,145 @@ describe('components/create_post', () => {
         expect(wrapper.state('showConfirmModal')).toBe(false);
     });
 
+    it('onSubmit test for @groups', () => {
+        const wrapper = shallowWithIntl(createPost());
+
+        wrapper.setProps({
+            allowReferencedGroups: [{
+                id: 'developers',
+                name: 'developers',
+            }],
+            channelMemberCountsByGroup: {
+                developers: {
+                    channel_member_count: 10,
+                    channel_member_timezones_count: 0,
+                }
+            }
+        });
+        wrapper.setState({
+            message: '@developers',
+        });
+
+        const form = wrapper.find('#create_post');
+        form.simulate('Submit', {preventDefault: jest.fn()});
+        expect(wrapper.state('showConfirmModal')).toBe(true);
+        expect(wrapper.state('memberNotifyCount')).toBe(10);
+        expect(wrapper.state('channelTimezoneCount')).toBe(0);
+        expect(wrapper.state('mentions')).toMatchObject(['@developers']);
+        wrapper.instance().hideNotifyAllModal();
+        expect(wrapper.state('showConfirmModal')).toBe(false);
+    });
+
+    it('onSubmit test for several @groups', () => {
+        const wrapper = shallowWithIntl(createPost());
+
+        wrapper.setProps({
+            allowReferencedGroups: [
+                {
+                    id: 'developers',
+                    name: 'developers',
+                },
+                {
+                    id: 'boss',
+                    name: 'boss',
+                },
+                {
+                    id: 'love',
+                    name: 'love',
+                },
+                {
+                    id: 'you',
+                    name: 'you',
+                },
+            ],
+            channelMemberCountsByGroup: {
+                developers: {
+                    channel_member_count: 10,
+                    channel_member_timezones_count: 0,
+                },
+                boss: {
+                    channel_member_count: 20,
+                    channel_member_timezones_count: 0,
+                },
+                love: {
+                    channel_member_count: 30,
+                    channel_member_timezones_count: 0,
+                },
+                you: {
+                    channel_member_count: 40,
+                    channel_member_timezones_count: 0,
+                },
+            }
+        });
+        wrapper.setState({
+            message: '@developers @boss @love @you',
+        });
+
+        const form = wrapper.find('#create_post');
+        form.simulate('Submit', {preventDefault: jest.fn()});
+        expect(wrapper.state('showConfirmModal')).toBe(true);
+        expect(wrapper.state('memberNotifyCount')).toBe(40);
+        expect(wrapper.state('channelTimezoneCount')).toBe(0);
+        expect(wrapper.state('mentions')).toMatchObject(['@developers', '@boss', '@love', '@you']);
```

Comment:
nit: might want to ad to the set a group with the name with a "-", for completeness (e.g. software-developers")

Reply:
Done